core/src/main/java/org/elasticsearch/common/xcontent/XContentParserUtils.java

```diff
@@ -36,34 +35,6 @@ private XContentParserUtils() {
     }
 
     /**
-     * Makes sure that current token is of type {@link XContentParser.Token#FIELD_NAME}
-     *
-     * @return the token
-     * @throws ParsingException if the token is not of type {@link XContentParser.Token#FIELD_NAME}
-     */
-    public static Token ensureFieldName(Token token, Supplier<XContentLocation> location) throws IOException {
-        return ensureType(Token.FIELD_NAME, token, location);
-    }
-
-    /**
-     * Makes sure that current token is of type {@link XContentParser.Token#FIELD_NAME} and the the field name is equal to the provided one
-     *
-     * @return the token
-     * @throws ParsingException if the token is not of type {@link XContentParser.Token#FIELD_NAME} or is not equal to the given
-     *                          field name
-     */
-    public static Token ensureFieldName(XContentParser parser, Token token, String fieldName) throws IOException {
-        Token t = ensureType(Token.FIELD_NAME, token, parser::getTokenLocation);
-
-        String current = parser.currentName() != null ? parser.currentName() : "<null>";
-        if (current.equals(fieldName) == false) {
-            String message = "Failed to parse object: expecting field with name [%s] but found [%s]";
-            throw new ParsingException(parser.getTokenLocation(), String.format(Locale.ROOT, message, fieldName, current));
-        }
-        return t;
-    }
-
-    /**
```

Comment:
sorry @tlrx while using these methods I changed them quite a bit, let me know what you think. I noticed one of them was unused, so I removed it.